File: easy/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function (nums) {
    const seen = new Set();

    for (const num of nums) {
        // We can exit early if duplicate is found
        if (seen.has(num)) return true;
        seen.add(num);
    }

    return false;
};

// Time complexity: O(n)
// Space complexity: O(n)
// where n is the length of nums

File: easy/0217-contains-duplicate.py
from typing import List

# Hash set
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False
    
# Hash set length
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(set(nums)) < len(nums)